Moved SimulationOptions into its own file. Added some Doxygen comments.

liangfok opened the pull request:
I was motivated to do this when working with the Kuka simulation library, which defines `SimulationOptions` but does call the `simulate()` method.

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/2952)
<!-- Reviewable:end -->

liangfok commented:
+@rpoyner-tri for feature review please.
+@sammy-tri for platform review please.

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2952#-:-KNhmqfuXL4AtoPlNI6y:535546338)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (presuming Jenkins passes, which seems likely)

---

Reviewed 1 of 3 files at r1.
Review status: 1 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2952#-:-KNhoRKfV0D9M3rQYAyc:1112054271)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Reviewed 2 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2952)*
<!-- Sent from Reviewable.io -->

rpoyner-tri commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[drake/systems/simulation_options.h, line 66 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/2952#-KNhp4GKY0lY_uv94t73:-KNhp4GKY0lY_uv94t74:-332976357) ([raw file](https://github.com/robotlocomotion/drake/blob/68b34d928e975bb463c3731a11fc22e5314d6532/drake/systems/simulation_options.h#L66)):*
> ```objc
>  * Defines the default simulation options.
>  */
> static const SimulationOptions default_simulation_options;
> ```

This is weird. What you have here is a global data definition in a header file. Stranger still, its scope is static. The result of of this will be to get one copy of the data in every translation unit, and none of them will be link-time public. Can we just delete this and rely on calling the default constructor instead?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2952)*
<!-- Sent from Reviewable.io -->

rpoyner-tri commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2952)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: 1 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/simulation_options.h, line 66 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/2952#-KNhp4GKY0lY_uv94t73:-KNhra8ggSYxE8-5OquE:723977643) ([raw file](https://github.com/robotlocomotion/drake/blob/68b34d928e975bb463c3731a11fc22e5314d6532/drake/systems/simulation_options.h#L66)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary>
> This is weird. What you have here is a global data definition in a header file. Stranger still, its scope is static. The result of of this will be to get one copy of the data in every translation unit, and none of them will be link-time public. Can we just delete this and rely on calling the default constructor instead?
</details>
Done. I deleted it and updated all references to it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2952)*
<!-- Sent from Reviewable.io -->

rpoyner-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 6 of 6 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2952#-:-KNht01AD3rnfUGtq_Le:1417088721)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Thanks. Just waiting on CI now.

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2952#-:-KNhuBf8_HIULmHa35wh:585891640)*
<!-- Sent from Reviewable.io -->
